### PR TITLE
Use Plone's registry instead of the portal_properties tool to store package configuration

### LINKED ISSUE #1: use plone.app.registry
we need to update the code to use plone.app.registry for the control panel.

this will break compatibility with Plone 3.x, but that has to be done at some point.

we will need also an upgrade step.

### PR DESCRIPTION
**Pull request by hvelarde**
This is needed to achieve future Plone 5 compatibility.

closes #1

### CODE REVIEW

**Comment by hvelarde** [comment]
@idgserpro I need help to review this

File: sc/social/like/plugins/facebook/browser.py [hunk 1]
```diff
@@ -108,6 +104,46 @@ def typebutton(self):
             self.width = '55px'
```
Comment (rodfersou) [inline sc/social/like/plugins/facebook/browser.py]: why not use ``button_count`` and ``box_count`` as value of the typebutton controlpanel option?

Comment (rodfersou) [inline sc/social/like/plugins/facebook/browser.py]: also, it is weird to calculate width into this property, I think this code could be extracted from here

**Comment by idgserpro** [comment]
@hvelarde is necessary run ```./bin/rebuild_i18n.sh```

**Comment by idgserpro** [comment]
@hvelarde a upgrade step is necessary.

**Comment by idgserpro** [comment]
> a upgrade step is necessary.

Sorry. It's in TODO.

**Comment by idgserpro** [comment]
@hvelarde is necessary change [this](https://github.com/collective/sc.social.like/blob/master/sc/social/like/profiles/default/controlpanel.xml#L7) to ```@@sociallike-settings```.

Don't forget this in upgrade step.

**Comment by hvelarde** [comment]
@idgserpro please comment directly on the code: the upgrade step is almost ready; I'll leave i18n outside of this PR to avoid pollution.

File: sc/social/like/plugins/facebook/browser.py (hunk 1 above)
Comment (hvelarde) [inline sc/social/like/plugins/facebook/browser.py]: @rodfersou I know, but I'm trying to move the less I can to avoid regressions.

**Comment by hvelarde** [comment]
@rodfersou @idgserpro ready to be tested; do not merge yet.

**Comment by idgserpro** [comment]
@hvelarde if a portal had the sc.social.like 2.4.1, and upgraded to this branch, the error below occurs  to start the instance:

```bash
2016-06-08 18:04:01 INFO ZServer HTTP server started at Wed Jun  8 18:04:01 2016
	Hostname: 0.0.0.0
	Port: 8080
Traceback (most recent call last):
  File "/tmp/sc.social.like/parts/instance/bin/interpreter", line 268, in <module>
    execfile(__file__)
  File "/home/user/cache-python/eggs/Zope2-2.13.24-py2.7.egg/Zope2/Startup/run.py", line 76, in <module>
    run()
  File "/home/user/cache-python/eggs/Zope2-2.13.24-py2.7.egg/Zope2/Startup/run.py", line 22, in run
    starter.prepare()
  File "/home/user/cache-python/eggs/Zope2-2.13.24-py2.7.egg/Zope2/Startup/__init__.py", line 92, in prepare
    self.startZope()
  File "/home/user/cache-python/eggs/Zope2-2.13.24-py2.7.egg/Zope2/Startup/__init__.py", line 268, in startZope
    Zope2.startup()
  File "/home/user/cache-python/eggs/Zope2-2.13.24-py2.7.egg/Zope2/__init__.py", line 47, in startup
    _startup()
  File "/home/user/cache-python/eggs/Zope2-2.13.24-py2.7.egg/Zope2/App/startup.py", line 118, in startup
    load_zcml()
  File "/home/user/cache-python/eggs/Zope2-2.13.24-py2.7.egg/Zope2/App/startup.py", line 52, in load_zcml
    load_site()
  File "/home/user/cache-python/eggs/Zope2-2.13.24-py2.7.egg/Zope2/App/zcml.py", line 46, in load_site
    _context = xmlconfig.file(site_zcml)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 649, in file
    include(context, name, package)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 210, in feed
    self._parser.Parse(data, isFinal)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 352, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 706, in finish
    actions = self.handler(context, **args)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 210, in feed
    self._parser.Parse(data, isFinal)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 352, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 706, in finish
    actions = self.handler(context, **args)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 210, in feed
    self._parser.Parse(data, isFinal)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 352, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 706, in finish
    actions = self.handler(context, **args)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 210, in feed
    self._parser.Parse(data, isFinal)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 352, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 706, in finish
    actions = self.handler(context, **args)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 210, in feed
    self._parser.Parse(data, isFinal)
  File "/home/user/pythons/python279/lib/python2.7/xml/sax/expatreader.py", line 352, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 705, in finish
    args = toargs(context, *self.argdata)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 1397, in toargs
    args[str(name)] = field.fromUnicode(s)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/fields.py", line 137, in fromUnicode
    value = self.context.resolve(name)
  File "/home/user/cache-python/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 198, in resolve
    return __import__(mname+'.'+oname, *_import_chickens)
  File "/tmp/sc.social.like/sc/social/like/upgrades/v3040/__init__.py", line 5, in <module>
    from sc.social.like.interfaces import ISocialLikeSettings
zope.configuration.xmlconfig.ZopeXMLConfigurationError: File "/tmp/sc.social.like/parts/instance/etc/site.zcml", line 15.2-15.55
    ZopeXMLConfigurationError: File "/tmp/sc.social.like/parts/instance/etc/package-includes/002-sc.social.like-configure.zcml", line 1.0-1.58
    ZopeXMLConfigurationError: File "/tmp/sc.social.like/sc/social/like/configure.zcml", line 12.4-12.36
    ZopeXMLConfigurationError: File "/tmp/sc.social.like/sc/social/like/profiles.zcml", line 26.4-26.35
    ZopeXMLConfigurationError: File "/tmp/sc.social.like/sc/social/like/upgrades/configure.zcml", line 12.4-12.32
    ImportError: cannot import name ISocialLikeSettings
```

**Comment by rodfersou** [comment]
I got the same exception here

**Comment by rodfersou** [comment]
I think the problem is with some changes in the namespace

**Comment by rodfersou** [comment]
The problem is with an old ``interfaces`` folder that keeps in the directory

**Comment by rodfersou** [comment]
There are a ``controlpanel`` folder that should be removed too

**Comment by rodfersou** [comment]
before I upgraded I filled randomly the configlet with some data, and when I run the upgrade step I got this exception:

```
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module Products.GenericSetup.tool, line 1034, in manage_doUpgrades
  Module Products.GenericSetup.upgrade, line 166, in doStep
  Module sc.social.like.upgrades.v3040, line 38, in migrate_settings_to_registry
  Module plone.registry.recordsproxy, line 45, in __setattr__
  Module plone.registry.registry, line 46, in __setitem__
  Module plone.registry.record, line 81, in _set_value
  Module zope.schema._bootstrapfields, line 182, in validate
  Module zope.schema._field, line 181, in _validate
  Module zope.schema._bootstrapfields, line 309, in _validate
  Module zope.schema._bootstrapfields, line 209, in _validate
WrongType: (u'rodfersou', <type 'str'>, 'value')
```

**Comment by rodfersou** [comment]
I did a fix for the upgrade step, but after I run it, the like buttons didn't show up in the welcome page

**Comment by rodfersou** [comment]
And I'm getting this warning when I start the instance:

```
2016-06-08 18:50:39 WARNING OFS.Uninstalled Could not import class 'ISocialLikeLayer' from module 'sc.social.like.interfaces.socialikes'
```

**Comment by rodfersou** [comment]
Looks like if the social like viewlet disapeared

File: sc/social/like/upgrades/v3040/__init__.py [hunk 2]
```diff
@@ -35,10 +35,10 @@ def migrate_settings_to_registry(setup_tool):
             settings.fbaction = old_props.fbaction
         except ConstraintNotSatisfied:  # empty string
             settings.fbaction = u'like'  # use default
-        settings.facebook_username = old_props.fbadmins
-        settings.facebook_app_id = old_props.fbapp_id
+        settings.facebook_username = old_props.fbadmins.encode('utf-8')
```
Comment (hvelarde) [inline sc/social/like/upgrades/v3040/__init__.py]: @rodfersou we do we need this?

**Comment by hvelarde** [comment]
@idgserpro issues resolved in 965505e9a8f783a5629771db89725d52a606f7e3.

@rodfersou before merging, please squash everything **except for the last commit that must remain separated**.

**Comment by rodfersou** [comment]
@hvelarde I'm not authorized to merge the request:
github_authorized

**Comment by idgserpro** [comment]
Guys do not forget the i18n.

**Pull request merged**